RESTserver/src/main/java/com/softserve/if072/restservice/service/AnalyticsService.java

```diff
@@ -65,8 +67,90 @@ public ProductStatistics getProductStatistics(int productId) throws NotEnoughDat
             int purchasingDatesSize = productStatistics.getPurchasingProductDates().length;
             productStatistics.setLastPurchasingDate(productStatistics.getPurchasingProductDates()[purchasingDatesSize - 1]);
         }
+        prepareDataForCharts(productStatistics);
         LOGGER.info(successfullyOperation, productId);
 
         return productStatistics;
     }
+
+    private static void prepareDataForCharts(ProductStatistics productStatistics) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for (double productUsingSpeed : productStatistics.getProductUsingSpeeds()) {
+            stringBuilder.append(String.format("%.1f ", productUsingSpeed));
+        }
+        String stringData = stringBuilder.toString().trim();
+        stringData = stringData.replace(',', '.');
+        productStatistics.setProductUsingSpeedsForChart(stringData);
+        productStatistics.setProductUsingSpeeds(null);
+
+        convertAmountsAndDatesDataFromArrayToString(productStatistics, Action.USED);
+        convertAmountsAndDatesDataFromArrayToString(productStatistics, Action.PURCHASED);
+    }
+
+    private static void convertAmountsAndDatesDataFromArrayToString(ProductStatistics productStatistics, Action action) {
+        int[] amounts;
+        Timestamp[] dates;
+        StringBuilder stringBuilderAmounts = new StringBuilder();
+        StringBuilder stringBuilderDates = new StringBuilder();
+        if (action == Action.USED) {
+            amounts = productStatistics.getUsingProductAmounts();
+            dates = productStatistics.getUsingProductDates();
+        } else {
+            amounts = productStatistics.getPurchasingProductAmounts();
+            dates = productStatistics.getPurchasingProductDates();
+        }
+
+        if (amounts != null) {
+            int i;
+            for (i = 1; i < dates.length; i++) {
+                int totalAmountPerDay = amounts[i - 1];
+                if (isTheSameDay(dates[i], dates[i - 1])) {
+                    totalAmountPerDay += amounts[i];
+                    i++;
+                    while (i < dates.length) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)